src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs

```diff
@@ -64,15 +70,26 @@ public async Task<long> FindPersonIdByUserOidAsync(string userOid)
             string lastName, 
             string userName)
         {
-            var shortName = userName?.Substring(0, userName.IndexOf('@', OrdinalIgnoreCase));
+            var shortName = string.IsNullOrEmpty(userName) || !userName.Contains("@")
+                ? string.Empty
+                : userName.Substring(0, userName.IndexOf('@')).ToUpper();
+
+            mobileNumber = string.IsNullOrEmpty(mobileNumber) ? string.Empty : mobileNumber.Replace(" ", string.Empty);
+            firstName = string.IsNullOrEmpty(firstName) ? string.Empty : firstName.ToUpper();
+            lastName = string.IsNullOrEmpty(lastName) ? string.Empty : lastName.ToUpper();
 
             return await _persons.Where(person =>
-                MobileNumberIsEqual(mobileNumber, person.MobilePhoneNumber)
-                || (firstName.Equals(person.FirstName, OrdinalIgnoreCase)
-                    && lastName.Equals(person.LastName, OrdinalIgnoreCase))
-                || string.Equals(shortName, person.UserName, OrdinalIgnoreCase)
-                || string.Equals(userName, person.UserName, OrdinalIgnoreCase)
-            ).ToListAsync();
+                    (person.MobilePhoneNumber != null &&
```

Comment:
I think the previous way of  writing this (aside from the  method) is quite a bit more readable? I especially miss the indentation.